docs/ops/deployment/overview.md

```diff
@@ -0,0 +1,358 @@
+<img width="100%" src="{% link fig/deployment_overview.svg %}" alt="Figure for Overview and Reference Architecture" />
+
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+      <th class="text-left" style="width: 25%">Component</th>
+      <th class="text-left" style="width: 50%">Purpose</th>
+      <th class="text-left">Implementations</th>
+    </tr>
+   </thead>
+   <tbody>
+        <tr>
+            <td>Flink Client</td>
+            <td>
+              Flink batch or streaming applications are compiled into a dataflow graph, which is submitted to the JobManager.
+            </td>
+            <td>
+                <ul>
+                    <li><a href="">Command Line Interface</a></li>
+                    <li><a href="">REST Endpoint</a></li>
+                    <li><a href="">SQL Client</a></li>
+                    <li><a href="">Python REPL</a></li>
+                    <li><a href="">Scala REPL</a></li>
+                </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>JobManager</td>
+            <td>
+                JobManager is the name of the central work coordination component of Flink. It has implementations for different resource providers, which differ on high-availability, resource allocation behavior and supported job submission modes. <br />
+                JobManager <a href="">modes for job submissions</a>:
+                <ul>
+                    <li><b>Session Mode</b>: one JobManager instance manages multiple jobs sharing the same cluster of TaskManagers</li>
+                    <li><b>Application Mode</b>: runs the cluster exclusively for one job. The job main method (or client) gets executed on the JobManager.</li>
+                    <li><b>Per-Job Mode</b>: runs the cluster exclusively for one job. The job main method (or client) runs only prior to the cluster creation.</li>
+                </ul>
+            </td>
+            <td>
+                <ul>
+                    <li><a href="">Kubernetes</a></li>
+                    <li><a href="">YARN</a></li>
+                    <li><a href="">Mesos</a></li>
+                    <li><a href="">Standalone</a> (this is the barebone mode that requires just JVMs to be launched. Deployment with Docker, Docker Swarm / Compose, non-native Kubernetes and other models is possible through manual setup in this mode)
+                    </li>
+                </ul>
+            </td>
+        </tr>
+        <tr>
+            <td colspan="3" class="text-center">
+                <b>External Components</b> (all optional)
+            </td>
+        </tr>
+        <tr>
+            <td>High Availability Service Provider</td>
+            <td>
+                Flink’s JobManager supports a high availability mode, where multiple JobManagers participate in a leader election process, resulting in an active JobManager and potentially multiple standby JobManagers, allowing for fast failover in case the active JobManager is lost.
```

Comment:
Maybe: Flink's JobManager can be run in high availability mode which allows Flink to recover from JobManager faults. In order to failover faster, multiple standby JobManagers can be started to act as backups.